presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -591,7 +706,7 @@ public BasicStageExecutionStats getBasicStageStats()
     public StageInfo getStageInfo()
     {
         ListMultimap<StageId, SqlStageExecution> stageExecutions = getStageExecutions();
-        return buildStageInfo(plan, stageExecutions);
+        return buildStageInfo(plan.get(), stageExecutions);
```

Comment:
This is likely to break statistics

CC: @rschlussel

Reply:
@pguofb is looking at that next. Right now the info in the live plan will be incorrect, but the final plan in the querycompletedevent and the stage info will be correct.